### PR TITLE
Blood Cult: Kills solo cult (summon ghost removed + shades cant use runes), adds a new feature to help with teamwork/coordination

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

Summon cult ghost and all related content has been removed. Spirit realm rune now only allows you to ascend as a spirit.

Shades can only use teleport runes now, and don't contribute towards invokers for runes that require more than 1 cultist.

Adds a new innate cult ability: Request Assistance. When used, allows you to request 1-3 cultists, and alerts cult as to this as well as where you are. Other cultists will then be able to respond to this request for assistance, giving them a pin pointer to your location and alerting other cult members. Each request lasts for 3 minutes, only one request can be active at a given time. ([Demo Video](https://youtu.be/NHYn-_l6xGY))

## Why It's Good For The Game

Cult is a team antagonist. As such, teamwork and cooperation should be required to win. Before this change, you could just solo cult by either killing someone and using a pocket shade(s) to convert a bunch of people, or using spirit realm. That's dead.

I could be convinced though to keep cult ghosts in and just remove their ability to use runes. The reason I decided to just remove was because A: its unbalanced (funny cult ghost spam), and B: gotta start somewhere to get rid of bloat. I'm not passionate about this point though, spam could probably just be solved by exponentially scaling damage.

An additional benefit of removing shades ability to use runes is that it will discourage just leaving people as a shade. Being stuck as a shade fucking sucks and is no fun, hopefully that will happen less after this.

Cult winrate would probably suffer a bit from just these changes before people got better at working together, so I added the Request Assistance button to help mitigate it somewhat. A lot of the time if you ask for help over cult comms no one will come, possibly due to assuming someone else will. Hopefully this will address that. 

## Changelog

:cl:
del: summon cult ghost and all related content is gone
balance: cult shades can now only use teleport runes
add: adds a new cult button: Request Assistance! By pressing it you can request the assistance of 1-3 cult members at a given location. Other cultists may then accept this call, which is broadcasted, and then are given a pin pointer leading them to your location.
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
gbp offer
Oh maintainers please accept this humble offering of virgin GBP

**Comment by JohnFulpWillard** [comment]
Cult always being the antag most likely to be reworked in big ways is my favorite thing about it, this is a really good change and I feel is a way to help simplify the overcomplicated cult mechanics. innate spell to ask for help seems way easier to use than the overcomplicated mechanics on getting a shard and offering ghosts control of a shade only for them to never play a round.

**Comment by tralezab** [comment]
spells to bypass cooperation in cult was a huge crutch, i'd rather the gamemode be better at getting people to work together than it going "ah well this shit never happens heres a spell"

**Comment by wesoda25** [comment]
I don't follow, are you referring to the spell I added?

**Comment by tralezab** [comment]
i'm telling you your pr is good

**Comment by Tanden22** [comment]
Quick question: do you have a way for a solo cultist to buff their ranks as a last ditch effort? Cause what i'm seeing here is if all the other cultists die/are deconverted/cryo, the last one is essentially boned with no way to win outside of arguably extreme circumstances (essentially making a golem).

**Comment by wesoda25** [comment]
Just make a construct.

**Comment by itseasytosee** [comment]
I quite like this, although I think it would get better reception if shades could still use runes.
I enjoy the idea of solo cultist being a difficult challenge rather than an inconvenience.

**Comment by SuperNovaa41** [comment]
The request assistance should probably be more impactful somehow, you know as well as I do this will be ignored 50%+ of the time on LRP

**Comment by wesoda25** [comment]
I would rather see how this does before deciding to buff it.

**Comment by datorangebottle** [comment]
While tackling one of my largest complaints about cult- building a base in space near the station and spamming cult ghosts onto it- this also removes something that annoyed me about cult: its inevitability. a handful of cultists + spirit realm could  turn back a lost cult round and summon nar'sie if the other objective had already been accomplished.
The only issue that needs to be tackled now is the funny red hand, because instant stun combat is bad.
Goodbye, cult ghosts. I will not miss you.

**Comment by SuperNovaa41** [comment]
> The only issue that needs to be tackled now is the funny red hand, because instant stun combat is bad

Good luck with that lmfao
You've gotta take oranges head on for that

**Comment by wesoda25** [comment]
In the big picture cult stun is a necessary evil. Yes it's not much fun to fight against, but given the current nature of the gamemode cult cannot operate without it. It would be like forcing security forces from 2018 to work with todays toolset. Sure, some people will be driven/robust enough to learn the new metas and manage to be competent, such as they did in security, however cult is a gamemode which you do not opt into, unlike security. So you're gonna get a bunch of people who aren't willing to invest the time getting shredded - which btw is the symptom of the major flaws in modern cult design: its like revs, but instead of utilizing features every player is acquainted with (the sandbox), it has its own sandbox which most do not care or have yet to adequately explore. This is why cults typically are just carried by 1-3 prominent members, and is an inherent flaw that I don't see being solved until either cult is either majorly simplified, made far more intuitive, and/or reworked so that it does not force itself on the playerbase at large (read: remove conversion and does not majorly derail the round).

In other words, short of major rework I can't see us living without it. See #50997 and #50522 for history.

**Comment by theta006** [comment]
time to only kill now because the other cultist is a cripple

**Comment by necromanceranne** [comment]
I am a little worried about bottlenecking issues with cult constructe taking too long to create, and this possible bottlenecking slowing cult down enough that they'll be lingering in low numbers for far longer (which might increase the loss rate to even higher than the staggering 80%ish loss rate it has now)

File: code/modules/antagonists/cult/cult_comms.dm [hunk 1]
```diff
@@ -79,6 +79,100 @@
...
+	team.cult_team.need_of_help = owner
+	for(var/datum/mind/cultist as anything in get_antag_minds(/datum/antagonist/cult))
+		if(cultist.current && cultist.current.stat != DEAD && cultist.current.client)
+			var/area/location = get_area(owner)
+			var/title = "Acolyte"
+			if(owner.mind && owner.mind.has_antag_datum(/datum/antagonist/cult/master))
+				title = "Master"
+			else if(!ishuman(owner))
+				title = "Construct"
+			to_chat(cultist.current, span_cultitalic("<b>[title] [findtextEx(owner.name, owner.real_name) ? owner.name : "[owner.real_name] (as [owner.name])"] has requested the aid of [number] cultists in the [location.name]! <a href=?src=[REF(src)];pledge_assistance=1>(Pledge Assistance)</b>"))
```
Comment (Fikou) [inline code/modules/antagonists/cult/cult_comms.dm:112]: the [location.name] is bad, either use [location.name] or just [location] or \the [location.name]

File: code/modules/antagonists/cult/cult_comms.dm [hunk 2]
```diff
@@ -79,6 +79,100 @@
...
+	var/number = tgui_alert(owner, "How many cultists do you need?",, list("0", "1", "2", "3"))
+	if(number == "0")
+		to_chat(owner, "You decide not to request help.")
+		return
+	team.cult_team.need_of_help = owner
+	for(var/datum/mind/cultist as anything in get_antag_minds(/datum/antagonist/cult))
+		if(cultist.current && cultist.current.stat != DEAD && cultist.current.client)
+			var/area/location = get_area(owner)
+			var/title = "Acolyte"
+			if(owner.mind && owner.mind.has_antag_datum(/datum/antagonist/cult/master))
```
Comment (Fikou) [inline code/modules/antagonists/cult/cult_comms.dm:108]:
```suggestion
			if(owner.mind?.has_antag_datum(/datum/antagonist/cult/master))
```

File: code/modules/antagonists/cult/cult_comms.dm (hunk 1 above)
Comment (Fikou) [reply code/modules/antagonists/cult/cult_comms.dm:112]: in the video for example you can see "requested assistance in the Engineering"

**Comment by Fikou** [comment]
> I am a little worried about bottlenecking issues with cult constructs taking too long to create

any job in the game that isnt assistant (who can just steal it from tool storage or aux base or whatever) can grab 50 iron from their departmental lathe to use the black hand on to create a construct

**Comment by necromanceranne** [comment]
> > I am a little worried about bottlenecking issues with cult constructs taking too long to create
> 
> any job in the game that isnt assistant (who can just steal it from tool storage or aux base or whatever) can grab 50 iron from their departmental lathe to use the black hand on to create a construct

50 iron is actually equivalent to a cyborg body with roundstart machines, so in some circumstances that metal disappears fast depending on pop present. Iron doesn't come in at a steady enough rate as is to keep up with station demands, so I can't imagine a cult is going to find any more success waiting on miners to deliver them metal. (Which is about 12-15 minutes in)

You'd maybe be able to get a few constructs deconstructing the hundreds of useless chairs on the station, but even that will hit its limits. To me, that says the cult has to metarush the metal supplies before the otherwise well meaning station members use it for practical everyday needs. The EVA supplies are pretty much the only ones I don't see touched most rounds, but its also the one with a motion sensor.

Not that cultists don't probably do that already, but I'm sure you get my meaning when I say it's not as simple a matter as just getting to something first. (And meta strats only weigh down cults skill curve even further)

**Comment by wesoda25** [comment]
Cult loss-rate is 62% as of 2 months ago. This decreases by about 10% with each successive milestone that is hit. I have no doubt that the loss-rate will increase after this PR, although hope it will return to normal with time.

I'm definitely in the camp that thinks cult is inherently flawed and needs a total rework, however unfortunately that is nowhere in sight. In the meantime I want to try to make it a healthier gamemode thats more fun for everyone. Assuming this is merged, if with time it becomes apparent that cult is dependent upon the crutches of solo cult, I'll probably revert those changes. 

As for the bottleneck thing - that's assuming you are purely killing instead of converting. Conversion should be encouraged over killing and sharding, so if issues exist with the latter I don't see it as much of an issue. Regardless I'm not convinced it will actually be one, each roundstart cultist is capable of producing ￼10 shells on their own. All it requires is simple resource management.

**Comment by Livrah** [comment]
Kills solo cult, adds solo murderboner cult.

**Comment by itseasytosee** [comment]
You can still make new converts with constructs. "Kills solo cult" is an exaggeration.

**Comment by wesoda25** [comment]
> Kills solo cult, adds solo murderboner cult.

This is... completely correct and something that I for whatever reason didn't give much thought to until now. If I want to actually rule out solo cult it'll require partially reworking how the offer rune works and I'm not sure if I want to get that intrusive with this PR. I need to give this some thought.

**Review by Twaticus — APPROVED** [review]

**Comment by ATH1909** [comment]
shades should still be allowed to aid with rune activation, imo, or else it becomes much harder to come back as the last remaining cultist. also, it'd make playing as a shade much more boring.

as for the summon ghost thing, here's a relevant PR of mine: https://github.com/tgstation/tgstation/pull/56876

**Comment by ATH1909** [comment]
also, you should at least let all cultists use teleport runes, or else that'll lead to awkwardness in lavaland, maint, and space bases

**Comment by wesoda25** [comment]
Mmk, few things that are bouncing around my head, would be interested in what others have to say.

This PR makes conversion more difficult, it does not affect sacrifice. Killing will likely become more common due to this (at least in the early parts of the round). My fear is that this will create an overall less enjoyable cult experience, regardless of whether or not it would be built on more healthy foundation (teamwork).

I could possibly address this by making sacrifice require two cultists. This would increase teamwork requirements across the board. My concern is that this could be detrimental to cult balance, and it would be basically impossible for cults to come back from the brink. The latter could be addressed by conditional requirements for invokers - if cult falls below a certain pop they can offer with lesser numbers again. Alternatively, less invokers would be required in certain areas - summon locations for instance. A rally would be difficult, but possible.

In all likeliness my concerns for both situations probably wouldn't look as bad as I fear, but regardless I think I'm left with 3 choices. 
1 - leave the PR as is, observe how cult adapts and see if killing/sacrifice becomes an issue. Conversion is a better investment than sacrifice, so it could not end up being a thing except when cult is being pushed hard. **Concern: creating a new, unfun meta**
2 - incorporate the aforementioned sacrifice changes. I'm not sure how much this would really change things - a dead/catatonic body or simplemob isnt going to run away from you, so there exists the luxury of being able to wait for someone to join you. I'm also not convinced that cult being able to come back from the brink is necessary **Concern: changes push cult too hard, players can't keep up**
3 - Just revert all the solo cult stuff, maybe leave shades being unable to use runes in (so as to discourage just leaving them to dry) and fix the cult ghost bugs **Problem: teamwork remains a suggestion for a team antag**

**Comment by tralezab** [comment]
go with number one, find out what happens, there is no time constraint for any of this!

**Comment by Coastfront** [comment]
Issue with the pinpointer idea: 1. This assumes that the person recieving the pinpointer, can even get in the room the conversion is happening in. If you are a cultist who doesent have maint acess, since cult bases are usually in maint, or if cultists are making an outpost to convert a department. Get fucked, You'll have to hack every door OR.
2. Why would the cult ever want to use this? For all intensive purposes, this is objectively a shittier method than just saying in cult chat "need help trying to convert clown in maint, i put a teleport rune down" and then a cutlist using the teleport spell/making their own rune and teleport to "CONVERTING CLOWN SEND HELP"
The "pls sneed hlep" rune idea seems like its just gonna be new player bait where new players get a convert, try to summon help, and then better players start yelling at them to make a teleport rune instead.

**Comment by wesoda25** [comment]
I'll add a bit of text to the request help button that they should consider making a teleport rune. The people accepting assistance already have it. It's intended to chiefl￼y help with coordination and address bystander effect where everyone assumes other people will help. As I said earlier I want to see how it goes before deciding to buff it. ￼

**Comment by KittyNoodle** [comment]
how to fix all problems: make summon cultist only need one person

**Comment by Iplayforss13** [comment]
I dont know, when you request for assistance, Its going to flood the cult chat, if you can strike a decent balance between the text, and maybe add a cooldown, but make the cooldown short, i don't want to screech over commune because my other cultists are too busy trying to convert other people because cult is so disorganised

**Comment by ATH1909** [comment]
> how to fix all problems: make summon cultist only need one person

the problem with that is that you can stun hand the cult's target, drag them into maint, begin carving a rune, and then be teleported halfway across the station by some random other cultist for their conversion of a duffel bag-wearing newfriend. actually, I guess that's a problem with summon cultist in general, ain't it? so yeah, making it only need one person would be nice, as would making it require consent (maybe something like that one xenobio crossbreed that you can resist out of the teleport from?).

**Comment by wesoda25** [comment]
Kay, this is ready. Did fikous review stuff, and made it so that cult shades can use teleport runes as ATHATH suggested. Gonna see how this goes, I think it's a valid concern that this might increase the amount of killing done. However conversion is inherently a better investment than sacrifice and the hope is that players are aware of this, so I'll see how this goes and update accordingly down the road if sacrifices become a problem.

**Comment by SethLafuente** [comment]
noooooo my ghost cultist

**Comment by wesoda25** [comment]
It's not a rune, just a button you press. _But_ making the button either spawn a teleport rune or turn the person requesting assistance into an object you can teleport to would actually be a good idea for a buff.

File: code/modules/antagonists/cult/cult.dm [hunk 3]
```diff
@@ -270,6 +274,7 @@
 	var/reckoning_complete = FALSE
 	var/cult_risen = FALSE
 	var/cult_ascendent = FALSE
+	var/need_of_help
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult.dm:277]: Doc-comment, including what type of variable this is.

File: code/modules/antagonists/cult/cult_comms.dm [hunk 4]
```diff
@@ -79,6 +79,101 @@
...
+
+/datum/action/innate/cult/request_aid/Activate()
+	var/datum/antagonist/cult/team = owner.mind.has_antag_datum(/datum/antagonist/cult,TRUE)
+	if(world.time < cooldown)
+		to_chat(owner, span_warning("You need to wait longer before requesting assistance again!"))
+		return
+	if(team.cult_team.need_of_help) //check to make sure no one else has already requested assistance
+		to_chat(owner, span_warning("Someone else is currently requesting assistance!"))
+		return
+	var/number = tgui_alert(owner, "How many cultists do you need?",, list("0", "1", "2", "3"))
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:99]: Nitpick, can we make these words? Single digits are harder to click on.

File: code/modules/antagonists/cult/cult_comms.dm [hunk 5]
```diff
@@ -79,6 +79,101 @@
 			var/link = FOLLOW_LINK(M, user)
 			to_chat(M, "[link] [my_message]")
 
+///a button for requesting cultists come to your location, lasts for 3 minutes (during which time no one else can request assistance)
+/datum/action/innate/cult/request_aid
+	name = "Request Assistance"
+	desc = "Request a number of cultists of your choosing to your location."
+	button_icon_state = "assistance"
+	check_flags = AB_CHECK_HANDS_BLOCKED|AB_CHECK_IMMOBILE|AB_CHECK_CONSCIOUS
+	var/cooldown = 0
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:88]: Use the cooldown defines.

File: code/modules/antagonists/cult/cult_comms.dm [hunk 6]
```diff
@@ -79,6 +79,101 @@
...
+	if(team.cult_team.need_of_help) //check to make sure no one else has already requested assistance
+		to_chat(owner, span_warning("Someone else is currently requesting assistance!"))
+		return
+	var/number = tgui_alert(owner, "How many cultists do you need?",, list("0", "1", "2", "3"))
+	if(number == "0")
+		to_chat(owner, "You decide not to request help.")
+		return
+	team.cult_team.need_of_help = owner
+	for(var/datum/mind/cultist as anything in get_antag_minds(/datum/antagonist/cult))
+		if(cultist.current && cultist.current.stat != DEAD && cultist.current.client)
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:105]: Early-return.

File: code/modules/antagonists/cult/cult_comms.dm [hunk 7]
```diff
@@ -79,6 +79,101 @@
...
+			if(owner.mind?.has_antag_datum(/datum/antagonist/cult/master))
+				title = "Master"
+			else if(!ishuman(owner))
+				title = "Construct"
+			to_chat(cultist.current, span_cultitalic("<b>[title] [findtextEx(owner.name, owner.real_name) ? owner.name : "[owner.real_name] (as [owner.name])"] has requested the assistance of [number] cultists in \the [location.name]! <a href=?src=[REF(src)];pledge_assistance=1>(Pledge Assistance)</b>"))
+			SEND_SOUND(cultist.current, sound(pick('sound/hallucinations/over_here2.ogg','sound/hallucinations/over_here3.ogg'),0,1,75))
+	to_chat(owner, span_boldnotice("You have requested assistance! Considering making a teleport rune with your dagger to help cultists reach you easier."))
+	cooldown = world.time + base_cooldown
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/reset_need_of_help,team.cult_team), base_cooldown, TIMER_STOPPABLE)
+	return
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:117]: No trailing return.

File: code/modules/antagonists/cult/cult_comms.dm [hunk 8]
```diff
@@ -79,6 +79,101 @@
...
+					var/title = "Acolyte"
+					if(usr.mind?.has_antag_datum(/datum/antagonist/cult/master))
+						title = "Master"
+					else if(!ishuman(usr))
+						title = "Construct"
+					to_chat(cultist.current, span_cultitalic("<b>[title] [findtextEx(user.name, user.real_name) ? user.name : "[user.real_name] (as [user.name])"]</b> has pledged their assistance!"))
+					SEND_SOUND(cultist.current, sound('sound/effects/ghost2.ogg',0,1,75))
+			user.apply_status_effect(/datum/status_effect/agent_pinpointer/cult_assistance)
+			to_chat(user, span_boldnotice("You have pledged your assistance! Follow the tracker to find the cultist and give them whatever help they need. Consider using teleport runes to get there faster!"))
+	return
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:145]: No trailing return

File: code/modules/antagonists/cult/cult_comms.dm [hunk 9]
```diff
@@ -79,6 +79,101 @@
...
+			to_chat(cultist.current, span_cultitalic("<b>[title] [findtextEx(owner.name, owner.real_name) ? owner.name : "[owner.real_name] (as [owner.name])"] has requested the assistance of [number] cultists in \the [location.name]! <a href=?src=[REF(src)];pledge_assistance=1>(Pledge Assistance)</b>"))
+			SEND_SOUND(cultist.current, sound(pick('sound/hallucinations/over_here2.ogg','sound/hallucinations/over_here3.ogg'),0,1,75))
+	to_chat(owner, span_boldnotice("You have requested assistance! Considering making a teleport rune with your dagger to help cultists reach you easier."))
+	cooldown = world.time + base_cooldown
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/reset_need_of_help,team.cult_team), base_cooldown, TIMER_STOPPABLE)
+	return
+
+///when a player clicks the "Pledge Assistance" link, they gain a tracker and the rest of cult is notified about it
+/datum/action/innate/cult/request_aid/Topic(href, href_list)
+	if(href_list["pledge_assistance"])
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:121]: Split the body of this into its own proc so that this is just:

```dm
if (href_list["pledge_assistance"])
    pledge_assistance(usr)
```

...or similar. That way it can all be better organized, and down an indent.

File: code/modules/antagonists/cult/cult_comms.dm [hunk 10]
```diff
@@ -79,6 +79,101 @@
...
+		if(istype(usr, /mob/living) && IS_CULTIST(usr.mind.current))
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:122]: Early-return. Also, just use `IS_CULTIST`, on its own, /mob/dead shouldn't be getting these anyway.

File: code/modules/antagonists/cult/cult_comms.dm [hunk 11]
```diff
@@ -79,6 +79,101 @@
...
+			var/mob/living/user = usr
+			var/datum/antagonist/cult/team = user.mind.has_antag_datum(/datum/antagonist/cult,TRUE)
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:124]: Use a named parameter for `TRUE`

File: code/modules/antagonists/cult/cult_comms.dm [hunk 12]
```diff
@@ -79,6 +79,101 @@
...
+			if(user == team.cult_team.need_of_help)
+				to_chat(user, span_warning("You can't answer your own request for assistance!"))
+				return
+			if(user.has_status_effect(/datum/status_effect/agent_pinpointer/cult_assistance)) //if you're already helping someone, you can't pledge again
+				to_chat(user, span_warning("You are already assisting someone!"))
+				return
+			for(var/datum/mind/cultist as anything in get_antag_minds(/datum/antagonist/cult))
+				if(cultist.current && cultist.current.stat != DEAD && cultist.current.client)
+					var/title = "Acolyte"
+					if(usr.mind?.has_antag_datum(/datum/antagonist/cult/master))
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:137]: Change these `usr` to `user`s, which you already have a var for, but when you split this out into a proc will need anyway.

File: code/modules/antagonists/cult/cult_comms.dm [hunk 13]
```diff
@@ -79,6 +79,101 @@
...
+				to_chat(user, span_warning("No one is in need of assistance right now!"))
+				return
+			if(user == team.cult_team.need_of_help)
+				to_chat(user, span_warning("You can't answer your own request for assistance!"))
+				return
+			if(user.has_status_effect(/datum/status_effect/agent_pinpointer/cult_assistance)) //if you're already helping someone, you can't pledge again
+				to_chat(user, span_warning("You are already assisting someone!"))
+				return
+			for(var/datum/mind/cultist as anything in get_antag_minds(/datum/antagonist/cult))
+				if(cultist.current && cultist.current.stat != DEAD && cultist.current.client)
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:135]: Early-return.

File: code/modules/antagonists/cult/cult_comms.dm [hunk 14]
```diff
@@ -79,6 +79,101 @@
...
+///locates the cult member in need of assistance
+/datum/status_effect/agent_pinpointer/cult_assistance/scan_for_target()
+	var/datum/antagonist/cult/team = owner.mind.has_antag_datum(/datum/antagonist/cult,TRUE)
+	scan_target = null
+	if(team.cult_team.need_of_help) //check to make sure someone has requested help
+		var/mob/living/target = team.cult_team.need_of_help
+		scan_target = target
+	else
+		owner.remove_status_effect(/datum/status_effect/agent_pinpointer/cult_assistance)
+	return
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:167]: No trailing return

File: code/modules/antagonists/cult/cult_comms.dm [hunk 15]
```diff
@@ -79,6 +79,101 @@
...
+
+///removes anyone in need of help, and wipes the pin pointer from anyone helping them. called once the duration of request assistance has run its course
+/proc/reset_need_of_help(datum/team/cult/team)
+	for(var/datum/mind/cultists in team.members)
+		if(cultists.current && cultists.current.stat != DEAD && cultists.current.client)
+			to_chat(cultists, span_warning("The request for assistance expires!"))
+	team.need_of_help = null
+	return
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:175]: No trailing return

File: code/modules/antagonists/cult/cult_comms.dm [hunk 16]
```diff
@@ -79,6 +79,101 @@
...
+	if(team.cult_team.need_of_help) //check to make sure someone has requested help
+		var/mob/living/target = team.cult_team.need_of_help
+		scan_target = target
+	else
+		owner.remove_status_effect(/datum/status_effect/agent_pinpointer/cult_assistance)
+	return
+
+///removes anyone in need of help, and wipes the pin pointer from anyone helping them. called once the duration of request assistance has run its course
+/proc/reset_need_of_help(datum/team/cult/team)
+	for(var/datum/mind/cultists in team.members)
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:171]: Change to `cultist`, as its only one.

Also, `as anything` IIRC.

File: code/modules/antagonists/cult/cult_comms.dm [hunk 17]
```diff
@@ -79,6 +79,101 @@
...
+	scan_target = null
+	if(team.cult_team.need_of_help) //check to make sure someone has requested help
+		var/mob/living/target = team.cult_team.need_of_help
+		scan_target = target
+	else
+		owner.remove_status_effect(/datum/status_effect/agent_pinpointer/cult_assistance)
+	return
+
+///removes anyone in need of help, and wipes the pin pointer from anyone helping them. called once the duration of request assistance has run its course
+/proc/reset_need_of_help(datum/team/cult/team)
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:170]: Make this a proc on `/datum/team/cult` instead of a global proc.

File: code/modules/antagonists/cult/runes.dm [hunk 18]
```diff
@@ -100,7 +101,7 @@ Runes can either be invoked by one's self or with many different cultists. Each
 		if(istype(user, /mob/living/simple_animal/hostile/construct/wraith/angelic) || istype(user, /mob/living/simple_animal/hostile/construct/juggernaut/angelic) || istype(user, /mob/living/simple_animal/hostile/construct/artificer/angelic))
 			to_chat(user, span_warning("You purge the rune!"))
 			qdel(src)
-		else if(construct_invoke || !IS_CULTIST(user)) //if you're not a cult construct we want the normal fail message
+		else if((istype(user, /mob/living/simple_animal/hostile/construct) && construct_invoke) || (istype(user, /mob/living/simple_animal/shade) && shade_invoke) || !IS_CULTIST(user)) //if you're not a cult construct/shade we want the normal fail message
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/runes.dm:104]:
```suggestion
		else if((isconstruct(user) && construct_invoke) || (isshade(user) && shade_invoke) || !IS_CULTIST(user)) //if you're not a cult construct/shade we want the normal fail message
```

File: code/modules/antagonists/cult/runes.dm [hunk 19]
```diff
@@ -781,109 +781,43 @@ structure_check() searches for nearby cultist structures required for the invoca
...
-/mob/living/carbon/human/cult_ghost/spill_organs(no_brain, no_organs, no_bodyparts) //cult ghosts never drop a brain
-	no_brain = TRUE
-	. = ..()
-
-/mob/living/carbon/human/cult_ghost/getorganszone(zone, subzones = 0)
-	. = ..()
-	for(var/obj/item/organ/brain/B in .) //they're not that smart, really
-		. -= B
-
+	affecting = user
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/runes.dm:790]: This is all bad but I'm pretty sure you didn't write it so whatever

File: code/modules/antagonists/cult/cult_comms.dm [hunk 20]
```diff
@@ -79,6 +79,101 @@
...
+		to_chat(owner, span_warning("You need to wait longer before requesting assistance again!"))
+		return
+	if(team.cult_team.need_of_help) //check to make sure no one else has already requested assistance
+		to_chat(owner, span_warning("Someone else is currently requesting assistance!"))
+		return
+	var/number = tgui_alert(owner, "How many cultists do you need?",, list("0", "1", "2", "3"))
+	if(number == "0")
+		to_chat(owner, "You decide not to request help.")
+		return
+	team.cult_team.need_of_help = owner
```
Comment (Mothblocks) [inline code/modules/antagonists/cult/cult_comms.dm:103]: This needs to be a weakref, especially if it can last for 3 minutes (which means a GC run can happen in between, IIRC).

You can ask me for help and/or look at other code with `WEAKREF` to learn how to use them.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
We'll get through this one

**Comment by Likteer** [comment]
> 
> 
> Kills solo cult, adds solo murderboner cult.

Replace cult with a team of heretics

**Comment by Draggeru** [comment]
How do you propose a cult that's been reduced to one cultist to recover? Cant team antag if you don't have a fuckin team. I also dont like soul stoning people, playing as a construct sucks ass.

**Comment by Draggeru** [comment]
Also additionally: People are saying people will put more value in fighting for converts instead of constructs. No. They fucking won't. You obviously don't know how murder-happy people get when it's cult. It's easier to kill someone and soulstone them than to waste spell charges to keep them restrained. ESPECIALLY when you're the last cultist. Remove the option for non-lethality and people are going to be lethal out of panic or sheer laziness. Why dont we fix the game mode instead of breaking it more? Like oh, I don't know? Making the stun hand not an instant catch-all drop? Maybe make it work more like a stun baton?

**Comment by wesoda25** [comment]
Feel free to read the rest of the comments where everything you just said was addressed

**Comment by itseasytosee** [comment]
> How do you propose a cult that's been reduced to one cultist to recover? Cant team antag if you don't have a fuckin team. I also dont like soul stoning people, playing as a construct sucks ass.

In short, a solo cultist has a few different ways of making a construct (ruined metal, twisted construction on a borg.) With that construct they can convert more cultists because constructs can still invoke.

**Comment by Draggeru** [comment]
> > How do you propose a cult that's been reduced to one cultist to recover? Cant team antag if you don't have a fuckin team. I also dont like soul stoning people, playing as a construct sucks ass.
> 
> In short, a solo cultist has a few different ways of making a construct (ruined metal, twisted construction on a borg.) With that construct they can convert more cultists because constructs can still invoke.

Again, I don't like making people constructs. The construct ends up having to stay in the cult base with fuck all to do while I, the already struggling cultist, struggle to get more converts. Playing as a construct sucks ass and I refuse to do it unless it's required. You somehow missed my entire point of how unfun being a construct is when the cult needs to completely stealth

**Comment by ExcessiveUseOfCobblestone** [comment]
> Cant team antag if you don't have a fuckin team

That should be the point of the mode imo, I dont like that you can carry even if your entire team is shit and continues to be shit for the duration of the round.

File: code/modules/antagonists/cult/cult_comms.dm (hunk 5 above)
Comment (wesoda25) [reply code/modules/antagonists/cult/cult_comms.dm:88]: wdym by cooldown defines? I don't see anything for /datum/action or /datum/action/innate. There's a /datum/action/cooldown proc but none of the other cult stuff makes use of that.

File: code/modules/antagonists/cult/cult_comms.dm (hunk 11 above)
Comment (wesoda25) [reply code/modules/antagonists/cult/cult_comms.dm:124]: Not sure what you mean by this or how I would go about it.

File: code/modules/antagonists/cult/cult_comms.dm (hunk 5 above)
Comment (Mothblocks) [reply code/modules/antagonists/cult/cult_comms.dm:88]: COOLDOWN_DECLARE, COOLDOWN_FINISHED, etc

File: code/modules/antagonists/cult/cult_comms.dm (hunk 11 above)
Comment (Mothblocks) [reply code/modules/antagonists/cult/cult_comms.dm:124]:
```suggestion
			var/datum/antagonist/cult/team = user.mind.has_antag_datum(/datum/antagonist/cult, check_subtypes = TRUE)
```

File: code/modules/antagonists/cult/cult_comms.dm (hunk 5 above)
Comment (necromanceranne) [reply code/modules/antagonists/cult/cult_comms.dm:88]: https://github.com/tgstation/tgstation/blob/76cee651a6480d846c07641df48794e6eedce1f0/code/__DEFINES/cooldowns.dm

Stuff found here

**Comment by optimumtact** [comment]
9 days to fix a merge conflict
